File: creational patterns/builder pattern/go example/carBuilder.go
package main

type carBuilder struct{
	car Car
}

func newCarBuilder() *carBuilder{
	return new(carBuilder)
}

func (cb *carBuilder) setDoor(door int){
	cb.car.door = door
}

func (cb *carBuilder) setId(id string){
	cb.car.id = id
}

func (cb *carBuilder) setEngine(engine string){
	cb.car.engine = engine
}

func (cb *carBuilder) setSpeed(speed int){
	cb.car.speed = speed
}

func (cb *carBuilder) build() *Car{
	return &cb.car
}

File: creational patterns/builder pattern/java example/CarBuilder.java
public class CarBuilder implements Builder{
    private String id;
    private int door;
    private String engine;
    private int speed;
    @Override
    public CarBuilder speed(int speed) {
        this.speed = speed;
        return this;
    }

    @Override
    public CarBuilder door(int door) {
        this.door = door;
        return this;
    }

    @Override
    public CarBuilder engine(String engine) {
        this.engine = engine;
        return this;
    }

    @Override
    public CarBuilder id(String id) {
        this.id = id;
        return this;
    }

    public Car build(){
        return new Car(id, door, engine, speed);
    }
    
    
}
